compiler+runtime/src/cpp/jank/runtime/obj/persistent_vector.cpp

```diff
@@ -242,6 +242,36 @@ namespace jank::runtime::obj
     }
   }
 
+  persistent_vector_ref persistent_vector::assoc(object_ref const key, object_ref const val) const
+  {
+    if(key->type != object_type::integer)
+    {
+      throw std::runtime_error{ "Key must be integer." };
+    }
+
+    auto const i(static_cast<size_t>(expect_object<integer>(key)->data));
+    auto const size(data.size());
+
+    if(i > size)
+    {
+      throw std::runtime_error{ "Index out of bounds." };
+    }
+
+    if(i == size)
```

Comment:
Since [`.set`](https://sinusoid.es/immer/containers.html#_CPPv2NK5immer6vector3setE9size_type10value_type) is `Undefined for index >= size()`, we need to check for this case explicitly and `push_back` instead.